include/pybind11/pytypes.h

```diff
@@ -885,7 +883,7 @@ class bytes;
 
 class str : public object {
 public:
-    PYBIND11_OBJECT_CVT(str, object, detail::PyUnicode_Check_Permissive, raw_str)
+    PYBIND11_OBJECT_CVT(str, object, PyUnicode_Check, raw_str)
```

Comment:
OK. Maybe getting rid of `raw_str` is then a different discussion, afterwards? Because that was a breaking change, right?

Reply:
> OK. Maybe getting rid of `raw_str` is then a different discussion, afterwards? Because that was a breaking change, right?

Yes. — It was a mistake that I tried changing `raw_str` before this PR has made it into master: too many changes at once.

Reply:
:+1: